src/modules/client/components/clientAlerts/ClientAlertCard.tsx

```diff
@@ -0,0 +1,113 @@
+import { Box } from '@mui/material';
+import { Stack } from '@mui/system';
+import TitleCard from '@/components/elements/TitleCard';
+import ClientAlert, {
+  ClientAlertProps,
+} from '@/modules/client/components/clientAlerts/ClientAlert';
+import { CreateClientAlertButton } from '@/modules/client/components/clientAlerts/CreateClientAlertDialog';
+import { ClientWithAlertFieldsFragment } from '@/types/gqlTypes';
+
+export enum AlertContext {
+  Client = 'Client',
+  Household = 'Household',
+}
+
+export enum AlertPriority {
+  high = 3,
+  medium = 2,
+  low = 1,
+}
+
+interface ClientAlertCardProps {
+  clients: ClientWithAlertFieldsFragment[];
+  alertContext: AlertContext;
+  shouldRenderFrame?: boolean;
+}
+
+const ClientAlertCard: React.FC<ClientAlertCardProps> = ({
+  clients,
+  alertContext = AlertContext.Client,
+  shouldRenderFrame = true,
+}) => {
+  const shouldShowClientName =
+    alertContext === AlertContext.Household && clients.length > 1;
+
+  const clientAlerts: ClientAlertProps[] = [];
+  clients.forEach((c) => {
+    if (c.access.canViewClientAlerts) {
+      c.alerts.forEach((a) => {
+        clientAlerts.push({
+          alert: a,
+          client: c,
+          shouldShowClientName: shouldShowClientName,
+        });
+      });
+    }
+  });
+
+  clientAlerts.sort((a, b) => {
+    if (AlertPriority[a.alert.priority] === AlertPriority[b.alert.priority]) {
+      return Date.parse(a.alert.createdAt) - Date.parse(b.alert.createdAt);
+    } else {
+      return AlertPriority[b.alert.priority] - AlertPriority[a.alert.priority];
+    }
+  });
+
+  const title = `${alertContext} Alerts (${clientAlerts.length})`;
+
+  if (!shouldRenderFrame) {
```

Comment:
After the most recent changes, enabling us to show client alerts in the enrollment dialog box, `ClientAlertCard` now kinda has 2 different 'jobs' -- and it doesn't always actually appear as a card -- and it does contain a little bit of repeated code. (lines 60-68 are almost identical to lines 96-106)

I thought about a couple of different ways to refactor this and didn't really decide that any of the ones I explored was actually better than the code as it exists now, but I would certainly be open to restructuring if you have suggestions!

Reply:
scratch this question for now -- after some more work on the dialog modal a refactor is even more needed so I'll take another stab at it.